Domains: Improve transfer out messaging

klimeryk opened the pull request:
Fixes #7090

#### Changes proposed in this Pull Request

The current transfer out view does not explain properly what are the implications of a transfer out and what limitations are put in place when it's in that state:

<img width="758" alt="Screenshot 2021-02-11 at 17 06 40" src="https://user-images.githubusercontent.com/3392497/107677769-8659f700-6c92-11eb-8a44-e037f11c3a6c.png">

This PR tries to address it using the copy proposed in https://github.com/Automattic/wp-calypso/issues/7090#issuecomment-242102230 and then improved by @wensco in https://github.com/Automattic/wp-calypso/pull/50007#issuecomment-779375246 ❤️ 

#### Testing instructions

Go to Upgrades -> Domains -> example.com -> Transfer domain -> Transfer to another registrar.

Depending on the domain type and its state, the screen should look:

Non-WWD domain that is locked:

<img width="751" alt="Screenshot 2021-02-22 at 11 48 16" src="https://user-images.githubusercontent.com/3392497/108704417-e6a72f00-7503-11eb-8ac5-e7bbb5b30fc6.png">

Non-WWD domain that is unlocked:

<img width="746" alt="Screenshot 2021-02-22 at 11 48 59" src="https://user-images.githubusercontent.com/3392497/108704478-fc1c5900-7503-11eb-892b-d3cee276d72a.png">

WWD domain that is locked and has privacy enabled:

<img width="748" alt="Screenshot 2021-02-22 at 11 49 42" src="https://user-images.githubusercontent.com/3392497/108704601-240bbc80-7504-11eb-8814-7948768d6804.png">

WWD domain that is unlocked and has privacy disabled (note that there's sometimes a delay caused by the WWD API with regards to disabling privacy protection, so it might just say it's unlocked for a minute or two):

<img width="757" alt="Screenshot 2021-02-22 at 12 21 09" src="https://user-images.githubusercontent.com/3392497/108707813-7b139080-7508-11eb-9033-872301a9c59a.png">

Make sure the links go to the correct views (name server change and contact info edit).

klimeryk committed:
Domains: Improve transfer out messaging

Fixes #7090

wensco commented:
Thanks so much for updating this - we have quite a few users who start a transfer and are then frustrated when they can't make DNS changes etc. 

I think the 3rd paragraph looks good! The second one is IMHO too broad and really hard to to accurately capture. 

I don't know of any TLDs that take 2 weeks to transfer. gTLDs are ~5 days, some cc's are instant whereas others are 7-8 days. Maybe it would be best to say something like "Most transfers are completed in a week or less; however depending on your domain it may take longer. 

For privacy, it's trickier. We should not remove privacy at all except for WWD domains. Only users who have WWD privacy (and therefore are not covered by GDPR) will have their contact info exposed publicly. In some cases with ccTLDs, it's already exposed.

wensco commented:
Following up on privacy...it would be great if we could have 3 main messages:
1. copy for domains using Key Systems and most ccTLDs* (exceptions for cc's like .uk that already have specific messaging)
2. WWD domains with an active privacy sub
3. WWD domains without an active privacy sub

If it's too hard to split the WWD domains into 2 groups based on privacy subscriptions, then I'd rather err on the side of displaying a full warning and use the WWD messaging above for all WWD domains.

**1. Copy for domains using Key Systems and most ccTLDs**
For these domains I would change the copy to something like this (keeping all of the relevant links to support docs/setting etc):

> To transfer your domain, we must unlock it. It will remain unlocked until the transfer is canceled or completed. [Learn More.[(https://wordpress.com/support/move-domain/transfer-domain-registration/) 
Most transfers can be completed in a week or less; however depending on your domain it may take longer. Your domain will continue to work, but **domain settings such as name servers and contact information cannot be changed during the transfer**. If you need to make changes, please do so before starting the transfer process.

**2. WWD domains with an active privacy sub**
Suggested copy:
>To transfer your domain, we must unlock it and remove Privacy Protection. Your contact information will be publicly available during the transfer period. The domain will remain unlocked and your contact information will be publicly available until the transfer is canceled or completed. [Learn More.[(https://wordpress.com/support/move-domain/transfer-domain-registration/) 
Most transfers can be completed in a week or less; however depending on your domain it may take longer. Your domain will continue to work, but **domain settings such as name servers and contact information cannot be changed during the transfer**. If you need to make changes, please do so before starting the transfer process.

**3. WWD domains without an active privacy sub**
We should use the same messaging as item 1.

klimeryk committed:
Update the messaging for unlocked and locked domains, including WWD case

a8ci18n commented on client/my-sites/domains/domain-management/transfer/transfer-out/locked.jsx:
```diff
@@ -47,23 +46,25 @@ class Locked extends React.Component {
 							? translate(
-									'To transfer your domain, we must unlock it and remove Privacy Protection. ' +
-										'Your contact information will be publicly available during the transfer period.'
+									'To transfer your domain, we must unlock it and remove Privacy Protection. Your contact information will be publicly available during the transfer period. The domain will remain unlocked and your contact information will be publicly available until the transfer is canceled or completed.'
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'To transfer your domain, we must unlock it and remove Privacy Protection. Your contact information will be publicly available during the transfer period.' )` *ES Score: 8*

wensco commented:
Looking good! Quick clarification for WWD with privacy disabled. Is this a different case from WWD domains that do not have privacy at all? 

The reason why I'm asking is because for GDPR -protected registrants, WWD does not support privacy but their contact info is not publicly exposed. 

It's a small detail to be sure (though as a Verified Nitpicker™ I'm sure you approve...) but I wouldn't want someone to get upset with us over exposing their personal info when we are not actually doing so.

stephanethomas commented on client/my-sites/domains/domain-management/transfer/transfer-out/warning.jsx:
```diff
@@ -0,0 +1,40 @@
+			) }
+		</p>
+	);
+}
```
Minor but it may be a good idea to document props with `propTypes` (here is [an example](https://github.com/Automattic/wp-calypso/blob/76ae9577190269d7b3fb36fc45628e208189ea05/client/components/data/query-sites/index.jsx#L92-L100) for a function).

klimeryk replied:
Thanks, that's a good suggestion 👍

stephanethomas commented on client/my-sites/domains/domain-management/transfer/transfer-out/warning.jsx:
```diff
@@ -0,0 +1,40 @@
+	return (
+		<p>
+			{ translate(
+				'Most transfers can be completed in a week or less; however depending on your domain it may take longer. Your domain will continue to work, but {{strong}}domain settings such as {{nameserversLink}}name servers{{/nameserversLink}} and {{contactInformationEditLink}}contact information{{/contactInformationEditLink}} cannot be changed during the transfer.{{/strong}} If you need to make changes, p ...
```
I'm certainly splitting hairs but I believe the closing `{{/strong}}` should be located before the `.` at the end of the sentence.

klimeryk replied:
Yeah, it felt wrong to me too - thanks for confirming my fears 😬 I've moved the dot outside `strong` 👍

stephanethomas commented on client/my-sites/domains/domain-management/transfer/transfer-out/locked.jsx:
```diff
@@ -47,23 +46,25 @@ class Locked extends React.Component {
+									'To transfer your domain, we must unlock it. It will remain unlocked until the transfer is canceled or completed.'
+							  ) }{ ' ' }
 						<a href={ TRANSFER_DOMAIN_REGISTRATION } target="_blank" rel="noopener noreferrer">
 							{ translate( 'Learn More.' ) }
```
Not related to your changes but I'm tempted to think that `Learn More.` should be written `Learn more.` given it is used at the end of a paragraph.

klimeryk replied:
Makes sense 👍

stephanethomas reviewed (approved):
I've tested it, and it worked great. I left a few minor suggestions with the hope to become a Verified Nitpicker™ as well 😁. I will finish by saying that I'm not a big fan of having the `Learn More.` link on its own:

<img width="746" alt="Screenshot 2021-02-17 at 15 15 47" src="https://user-images.githubusercontent.com/3392497/108230183-62862d80-7138-11eb-938e-28386d232bf9.png">

I feel it should either be located at the end of the previous paragraph, or rewritten to provide more context (e.g. `Learn more about domain transfers.`).

klimeryk committed:
Address feedback from code review

klimeryk committed:
Prevent a fatal failure when we don't have domains loaded

a8ci18n commented on client/my-sites/domains/domain-management/transfer/transfer-out/warning.jsx:
```diff
@@ -0,0 +1,50 @@
+				}
+			) }{ ' ' }
+			<a href={ TRANSFER_DOMAIN_REGISTRATION } target="_blank" rel="noopener noreferrer">
+				{ translate( 'Learn more about domain transfers.' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Learn more about Guided Transfers' )` *ES Score: 7*

klimeryk commented:
Thanks for the careful review, @stephanethomas! May the God of Nitpicking smile upon you! 😄 Seriously, though - good points all around 🙇 

> I feel it should either be located at the end of the previous paragraph, or rewritten to provide more context (e.g. Learn more about domain transfers.).

Good point - I did not notice that in one of the cases the `Learn more` link got isolated like that. I've moved it to the warning component (since it makes a logical sense to show it there) and updated it to say `Learn more about domain transfers` 👍

klimeryk commented on client/my-sites/domains/domain-management/transfer/transfer-out/index.jsx:
```diff
@@ -102,7 +102,11 @@ class Transfer extends React.Component {
+		return (
+			! this.props.wapiDomainInfo.hasLoadedFromServer ||
+			this.props.isRequestingSiteDomains ||
+			! this.props.hasSiteDomainsLoaded
```
This fixes an unrelated bug that caused this whole screen to go blank when we're loading from empty/fresh state.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5543113

Thank you @klimeryk for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5543113).
